### PR TITLE
(BOLT-410) Allow descriptions for run_* functions and CLI

### PR DESCRIPTION
**Pull request by nicklewis**
The `run_task()`, `run_script()`, `run_command()`, and `file_upload()`
functions can now take a positional "description" argument after the target. If
supplied, the description will be logged at notice level when running that
step, and will also be set as the job description in orchestrator.

There is also now a `--description` argument which can be used to specify the
description for a one-off job from the CLI. That description will similarly be
logged and sent to orchestrator.

### CODE REVIEW

File: lib/bolt/executor.rb [hunk 1]
```diff
@@ -98,6 +98,7 @@ def batch_execute(targets)
     end
 
     def run_command(targets, command, options = {}, &callback)
+      @logger.notice(options['_description']) if options.key?('_description')
```
Comment (adreyer) [inline lib/bolt/executor.rb]: I think this should just be worked into the next message and follow the verbosity of it.

**Comment by adreyer** [comment]
The ticket in the commit message should be 410 not 140 too

**Comment by nicklewis** [comment]
I'm working on updating the log messages now. This is what I have with descriptions:

image

And without descriptions:

image

That seem reasonable?

**Comment by nicklewis** [comment]
I think it's actually more difficult to read if I replace the "finished" message with a generic one:

image

**Comment by nicklewis** [comment]
Updated.

**Pull request merged**